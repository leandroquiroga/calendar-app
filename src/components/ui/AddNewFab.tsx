import React from 'react'
import { useDispatch } from 'react-redux'
import { uiOpenModal } from '../../actions/actions';

export const AddNewFab = (): JSX.Element => {

  const dispatch = useDispatch();
  // Abre el modal 
  const handleOpenModal = () => dispatch(uiOpenModal())
  
  return (
    <button
      className='btn btn-primary shadow rounded-circle button_icon'
      onClick={handleOpenModal}
    >
      <i className="fa-solid fa-plus"></i>
    </button>
  )
};