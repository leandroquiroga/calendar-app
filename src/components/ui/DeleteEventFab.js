import React from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { deleteEventNotes } from '../../actions/events';


export const DeleteEventFab = () => {

  const dispatch = useDispatch();
  const {activeEvent} = useSelector(state => state.calendar)

  // Elimina el evento
  const handleDeleteNotes = () => dispatch(deleteEventNotes(activeEvent.id))

  return (
    <button
      className='btn btn-danger rounded-circle shadow button_icon__delete'
      onClick={handleDeleteNotes}
    >
        <i className="fa-solid fa-trash-can"></i>
    </button>
  )
}
