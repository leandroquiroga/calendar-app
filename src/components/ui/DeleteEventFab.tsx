import React from 'react'
import { useDispatch } from 'react-redux';
import { eventStartDelete } from '../../actions/events';

export const DeleteEventFab = () => {

  const dispatch = useDispatch();

  // Elimina el evento
  const handleDeleteNotes = () => dispatch(eventStartDelete())

  return (
    <button
      className='btn btn-danger rounded-circle shadow button_icon__delete'
      onClick={handleDeleteNotes}
    >
      <i className="fa-solid fa-trash-can"></i>
    </button>
  )
};