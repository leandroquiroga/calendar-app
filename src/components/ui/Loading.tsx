import React from 'react'

export const Loading = () => {
  return (
    <section className='d-flex justify-content-center align-items-center min-vh-100 bg-primary'>
      <div className="lds-ring">
        <div></div>
        <div></div>
        <div></div>
        <div></div>
      </div>
    </section> 
  )
}