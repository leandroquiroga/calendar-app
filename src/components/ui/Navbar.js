import React from 'react'

export const Navbar = () => {
  return (
    <nav className='navbar navbar-dark bg-dark'>
      <section className='container-fluid'>
        <span className='text-white'>Hola, Leandro !</span>
        <button className='btn btn-danger p-2 rounded'>
          <i className="fa-solid fa-arrow-right-from-bracket"></i>
          <span> Logout </span>
        </button>
      </section>
    </nav>
  )
}
