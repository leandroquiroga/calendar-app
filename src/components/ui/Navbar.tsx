import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startUserLogut } from '../../actions/auth';

export const Navbar = (): JSX.Element => {
  
  const { auth } = useSelector(state => state);
  const dispatch = useDispatch();

  const handleSingOut = () => dispatch(startUserLogut());
  
  return (
    <nav className='navbar navbar-primary bg-primary'>
      <section className='container-fluid'>
        <span className='text-white text-capitalize'>Hola, {auth.name} !</span>
        <button
          className='btn btn-outline-light p-2 rounded'
          onClick={handleSingOut}
        >
          <i className="fa-solid fa-arrow-right-from-bracket"></i>
          <span> Logout </span>
        </button>
      </section>
    </nav>
  )
};