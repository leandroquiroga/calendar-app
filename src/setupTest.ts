import '@testing-library/jest-dom';
import Enzyme from 'enzyme';
import ReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer,  } from 'enzyme-to-json';

Enzyme.configure({ adapter: new ReactAdapter() });

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}) as any)