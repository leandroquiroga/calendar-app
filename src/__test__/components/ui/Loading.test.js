import React from "react";
import { shallow } from "enzyme";
import { Loading } from "../../../components/ui/Loading";

import '@testing-library/jest-dom';
import '../../../setupTest';


describe('Test on <Loading />', () => {

  test('this test should make a snapshot with <Loading /> correctly', () => {
    const wapper = shallow(<Loading />);

    expect(wapper).toMatchSnapshot();
  });
});