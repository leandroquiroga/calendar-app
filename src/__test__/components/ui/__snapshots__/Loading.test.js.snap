// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Loading /> this test should make a snapshot with <Loading /> correctly 1`] = `
<section
  className="d-flex justify-content-center align-items-center min-vh-100 bg-primary"
>
  <div
    className="lds-ring"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</section>
`;
